components/script/dom/htmlformelement.rs

```diff
@@ -704,8 +728,28 @@ impl HTMLFormElement {
                 },
             }
         }
-        // Step 5
-        ret
+
+        let window = window_from_node(self);
+
+        // Step 6
+        let form_data = FormData::new(Some(ret.clone()), &window.global());
```

Comment:
I can udnerstand why the testcase failed now but I'm still trying to figure out how to avoid the `clone` here 🤔 Would it be a right way to pass the first argument of `FormData::new` with `Option<&Vec<FormDatum>>`?

Reply:
I expect you want to pass `ret` itself, then add an API to FormData to extract it again later and return that value.

Reply:
Ah! Yeah! I was thinking from constructor perspective 😂 That must be a good solution to extract it again! Thanks!

Reply:
Just found there's an existing API to extract the datums.

https://github.com/servo/servo/blob/192fbafd11126744a299558d14c4b2c63e3cd2ca/components/script/dom/formdata.rs#L197-L204

One more question for this; not sure why there're two `impl FormData` but maybe it's better to combine them? 🤔

Reply:
Combining them is fine.